### ISSUE TITLE
Webview version is too old (null) Kindle Fire HD 10 (7th Gen)

### ISSUE THREAD

**Issue by timbze** [AndBible/and-bible#1139]
https://support.andbible.org/scp/tickets.php?id=577
The bug report reports I have version 84 of webview, but for some reason the code check gets null.

Device info:
And Bible version: 3.4.527-beta#1eb8c92 (built 12/06/21 06:29:58)
Android version: 5.1.1
Android SDK version: 22
Manufacturer: Amazon
Model: KFSUWI
Storage Mb free: 17649
WebView version: 84.ship_v84-93-Arcanine.4147.125.180
SQLITE version: 3.8.6.1

**Comment by tuomas2** [comment]
Hmm. Similar issues can happen on all devices that do not contain standard Google's WebView (probably those devices that do not have Google Play).

**Comment by timbze** [comment]
The bug report has this: `WebView version: 84.ship_v84-93-Arcanine.4147.125.180`

So I'm guessing the version number could be parsed, but maybe would have to be done different from Google's webview...

**Comment by timbze** [comment]
Another issue, perhaps this should be a new issue... On this same device, I can only select 1 word. The arrows at the end of the selection do not come up. I can "Compare" and "Share/copy" with 1 word selected, but clicking "Bookmark" or "Bookmark verse" does not work, nothing happens.

Is this related to webview? Or is it because of old android version? Other apps can select more than 1 word though...

**Comment by tuomas2** [comment]
> Another issue, perhaps this should be a new issue... On this same device, I can only select 1 word. The arrows at the end of the selection do not come up. I can "Compare" and "Share/copy" with 1 word selected, but clicking "Bookmark" or "Bookmark verse" does not work, nothing happens.
> 
> Is this related to webview? Or is it because of old android version? Other apps can select more than 1 word though...

screenshot and perhaps new issue for this?

**Comment by tuomas2** [comment]
One thing I can't understand. why it says "null" in the title? The code is net/bible/android/view/activity/StartupActivity.kt:154 (checkWebView) 

and it looks that it gives null for  `WebViewCompat.getCurrentWebViewPackage(applicationContext)?.versionName` here, but proper version number in ErrorReportControl.kt:162 (createErrorText). 

Do you have the device permanently so you can do further tests later?

**Comment by tuomas2** [comment]
Parsing `84.ship_v84-93-Arcanine.4147.125.180` produces correctly 84, so parsing is not a problem here.

**Comment by tuomas2** [comment]
Can you try with tag build-527-kindlefire1 (build triggered)?

**Comment by timbze** [comment]
> Can you try with tag build-527-kindlefire1 (build triggered)?

Tested now and still has the same issue

**Comment by timbze** [comment]
> Do you have the device permanently so you can do further tests later?

Yes, it's my device

**Comment by tuomas2** [comment]
I don't know but I fear that that device might show correct info only after bibleview has been used (?). Then I don't know how to fix this. Could be also bug in WebViewCompat.

**Comment by timbze** [comment]
I think in the case when it's null it might be better to just skip the
check then?

**Comment by timbze** [comment]
Another note; the "webview too old" dialog shows twice on starting AB. Is that pointing to some other issue? Maybe it's no big deal.

Just tested on build 528
